src/storage.rs

```diff
@@ -112,20 +117,42 @@ enum Message {
 pub struct CelestiaConnection {
     pub client: WsClient,
     pub namespace_id: Namespace,
-    pub sync_target: Arc<Mutex<u64>>,
+    tx: Arc<mpsc::Sender<Message>>,
+    rx: Arc<tokio::sync::Mutex<mpsc::Receiver<Message>>>,
+}
+
+impl CelestiaConnection {
+    pub async fn new(connection_string: &String, auth_token: Option<&str>, namespace_hex: &String) -> CelestiaConnection {
+        let (tx, rx) = mpsc::channel(5);
```

Comment:
Well, this is a buffer value AFAIU. So that means that after 5 headers are received from celestia, if we are not processing them, the next message send (storage.rs:205) will block until the next one is processed. I'm not sure what a good value for this is so maybe we should extract it, but I think leaving it as 5 is fine for now as long as I add a TODO

Reply:
totally agree